### PR TITLE
rank up notification

### LINKED ISSUE #837: Ranked league notifications
Have the bot that is updating people's ranked roles based on their position on the leaderboard also tag these people in a message when they move up or down.
"Congrats XYZ for reaching the Silver league, knocking ZYX out of it"
Maybe without the "this player lost their position" for the first version?
In #tavern?

### PR DESCRIPTION
**Pull request by mquellhorst**
## Overview
Added a trigger in the backend which posts a message in the props channel when a player ranks up.
When the player goes down a rank nothing happens.

Closes #837 

## Implementation

Added a function to the utils package which determines the new rank is higher than the old rank.
The backend triggers the message to the props channel.

### CODE REVIEW

**Comment by vidvidvid** [comment]
Great stuff! :D Could you maybe just add an exclamation mark at the end of the message? It's a celebratory message after all :P   `Props to ${newPlayer?.username} for becoming ${newRank}`,

**Comment by mquellhorst** [comment]
Let's do some congratulations ;)

File: packages/backend/src/handlers/triggers/playerRankUpdated.ts [hunk 1]
```diff
@@ -104,6 +106,15 @@ export const playerRankUpdated = async (payload: TriggerPayload<Player>) => {
       console.warn(`Discord role associated with ${newRank} was not found!`);
     } else {
       await discordPlayer.roles.add([discordRoleForRank]);
+      // We should check whether this role is higher than the previous
+      if (playerRankedUp(oldPlayer?.rank, newPlayer?.rank)) {
+        const propsChannel = discordClient.channels.cache.get(
+          Constants.DISCORD_PROPS_CHANNEL_ID,
+        ) as TextChannel;
+        propsChannel.send(
+          `Props to ${newPlayer?.username} for becoming ${newRank}, congrats!`,
```
Comment (dan13ram) [inline packages/backend/src/handlers/triggers/playerRankUpdated.ts:115]: Could we tag the user here instead of printing just their username?

Comment (mquellhorst) [reply packages/backend/src/handlers/triggers/playerRankUpdated.ts:115]: Good one, we should. I'll add it thanks!

**Review by dysbulic — APPROVED** [review]
Couple suggestions on how to be more concise, but it looks good to me.

File: packages/backend/src/handlers/triggers/playerRankUpdated.ts [hunk 2]
```diff
@@ -106,6 +108,16 @@ export const playerRankUpdated = async (payload: TriggerPayload<Player>) => {
...
     } else {
       await discordPlayer.roles.add([discordRoleForRank]);
+
+      // We should check whether this role is higher than the previous
+      if (playerRankedUp(oldPlayer?.rank, newPlayer?.rank)) {
+        const propsChannel = discordClient.channels.cache.get(
+          Constants.DISCORD_PROPS_CHANNEL_ID,
+        ) as TextChannel;
+        propsChannel.send(
+          `Props to ${discordPlayer.toString()} for becoming ${newRank}, congrats!`,
```
[the remark this answers is not in the document]
Comment (mquellhorst) [reply packages/backend/src/handlers/triggers/playerRankUpdated.ts:118]: Very good question! 
I'm not sure but it's kinda hard to test out.
If you have a suggestion on how to test I'ld love to hear it

[the remark this answers is not in the document]
Comment (dysbulic) [reply packages/backend/src/handlers/triggers/playerRankUpdated.ts:118]: It looks like yes: https://stackoverflow.com/a/63184878/264008

**Pull request merged**